client/src/pages/Property.tsx

```diff
@@ -17,8 +16,20 @@ const API_URL = `${import.meta.env.VITE_API_SERVER_URL}/api/v1`;
  *
  */
 const Property = () => {
-  const { state } = useLocation();
   const toast = useToast();
+  const { state } = useLocation();
+  const [post, setPost] = useState<Post>({
+    authorId: "",
+    comments: [],
+    content: "",
+    createdAt: new Date(),
+    id: "",
+    postPhotos: [],
+    published: false,
+    updatedAt: new Date(),
+    title: "",
+    rating: 0,
+  });
```

Comment:
```ts
const [post, setPost] = useState<Post | null>(null);
```
Use this instead since you were implying an empty post.

Also we need loading state and error state as well:

```ts
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState<Error | null>(null);
```
Those should be set during the `useEffect`. The implementation is described in first part in https://dev.to/techcheck/custom-react-hook-usefetch-eid but I think it is good to just bring in the `useFetch` hook into this